gnomad_qc/v5/resources/basics.py

```diff
@@ -50,12 +88,249 @@ def get_checkpoint_path(
+    # Union all samples to exclude.
+    samples_to_exclude = bad_quality_ids.union(bad_genomic_samples).distinct()
+
+    # Filter out poor-quality samples.
+    vds = hl.vds.filter_samples(
+        vds, samples_to_exclude, keep=False, remove_dead_alleles=remove_dead_alleles
+    )
+
+    # Report final sample exclusion count.
+    n_samples_after = vds.variant_data.count_cols()
+    logger.info("Removed %d samples from VDS.", n_samples_before - n_samples_after)
+
+    if filter_samples:
+        logger.info(
+            "Filtering to %s samples...",
+            (
+                len(filter_samples)
+                if isinstance(filter_samples, list)
+                else filter_samples.count()
+            ),
+        )
+        vds = hl.vds.filter_samples(vds, filter_samples)
+
+    if naive_coalesce_partitions:
+        vds = hl.vds.VariantDataset(
+            vds.reference_data.naive_coalesce(naive_coalesce_partitions),
+            vds.variant_data.naive_coalesce(naive_coalesce_partitions),
+        )
+
+    # Apply interval filtering.
+    if filter_intervals and len(filter_intervals) > 0:
+        logger.info("Filtering to %s intervals...", len(filter_intervals))
+        if isinstance(filter_intervals[0], str):
+            filter_intervals = [
+                hl.parse_locus_interval(x, reference_genome="GRCh38")
+                for x in filter_intervals
+            ]
+        vds = hl.vds.filter_intervals(
+            vds, filter_intervals, split_reference_blocks=split_reference_blocks
+        )
+
+    # Apply partition filtering.
+    if filter_partitions and len(filter_partitions) > 0:
+        logger.info("Filtering to %s partitions...", len(filter_partitions))
+        vds = hl.vds.VariantDataset(
+            vds.reference_data._filter_partitions(filter_partitions),
+            vds.variant_data._filter_partitions(filter_partitions),
+        )
+
+    vmt = vds.variant_data
+
+    if filter_variant_ht is not None and split is False:
+        raise ValueError(
+            "Filtering to a specific set of variants is only supported when splitting"
+            " the VDS."
+        )
+    if split:
+        vmt = _split_and_filter_variant_data_for_loading(
+            vmt, filter_variant_ht, entries_to_keep, checkpoint_variant_data
+        )
+        # Don't need to filter entries again if already done in split function
+        entries_filtered_during_split = entries_to_keep is not None
+    else:
+        entries_filtered_during_split = False
+
+    if entries_to_keep is not None and not entries_filtered_during_split:
+        vmt = vmt.select_entries(*entries_to_keep)
+
+    if checkpoint_variant_data:
+        vmt = vmt.checkpoint(new_temp_file("vds_loading.variant_data", "mt"))
+
+    vds = hl.vds.VariantDataset(vds.reference_data, vmt)
+
+    return vds
+
+
+def get_invalid_aou_samples() -> hl.Table:
+    """
+    Import AoU genomic metrics and filter to invalid samples (ambiguous sex and failed coverage metrics).
```

Comment:
could you add more documentation about the thresholds and why this function is necessary?